### PR TITLE
Slight changes to spawners

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

Organizes the markers yaml. Also reworks the sprite component so we don't need custom sprites for the spawn icons, just layering. Changes the name of TrashSpawner.cs to RandomSpawner.cs because I think it's really good for general use. Adds a super-simple MobSpawner.cs.

:cl: Swept
- remove: Remove Mouse spawns on Saltern

### CODE REVIEW

File: Content.Server/GameObjects/Components/Markers/MobSpawnerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.Markers
+{
+    [RegisterComponent]
+    public class MobSpawnerComponent : Component, ISerializationHooks
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/Markers/MobSpawnerComponent.cs:12]: Don't need the hook as you're not calling the methods also file needs to be nullable

File: Content.Server/GameObjects/Components/Markers/MobSpawnerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/Markers/MobSpawnerComponent.cs:14]: Given the entity's being spawned once it'd be better for this to be resolved in Initialize to reduce memory usage.

If you really wanted you could also have the component nuke itself as well.

File: Resources/Prototypes/Entities/Markers/Spawners/Conditional/timed.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        - state: blue
+        - texture: Mobs/Aliens/Xenos/xeno.rsi/crit.png
+        - state: ai
+    - type: TimedSpawner
+      prototypes:
+        - XenoMob_Content
+      chance: 0.85
+      intervalSeconds: 30
+      minimumEntitiesSpawned: 2
+      maximumEntitiesSpawned: 4
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Markers/Spawners/Conditional/timed.yml:36]: Are these mapped in?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Should MobSpawner inherit from ConditionalSpawner or something? It seems like the functionality would already exist in the other spawners

At the very least call it something like "SimpleSpawner" given it's not mob exclusive.

File: Resources/Prototypes/Entities/Markers/Spawners/Conditional/timed.yml (hunk 3 above)
Comment (SweptWasTaken) [reply Resources/Prototypes/Entities/Markers/Spawners/Conditional/timed.yml:36]: > 
> 
> Are these mapped in?

No? I just found them.

**Comment by SweptWasTaken** [comment]
Deletes MobSpawnerComponent.cs because i'm stupid! ConditionalSpawner does the same exact thing!

**Review by metalgearsloth — APPROVED** [review]
YAML linter passed and tests are clickmap sooooo

**Comment by metalgearsloth** [comment]
Reverting until fixy fix

**Pull request merged**